File: backend/PrivateApi/Service/BatchLogger.cs
using CSLib.Lib;
using Dapper;
using PrivateApi.Domain.DTO;
using Repository.Repository;
using Repository.Repository.sql;

namespace PrivateApi.Service
{
    public class BatchLogger
    {
        private readonly IDatabase _posgre;

        public BatchLogger(IConfiguration Configuration)
        {
            _posgre = new PostgreSQL(ConnectionString.Get(Configuration));
        }

        public async Task<BatchConfigDTO> InsertStartLog(string batchName)
        {
            var param = new DynamicParameters();
            param.Add("batch_name", batchName);

            // idは自動採番
            await _posgre.Execute(BatchLogSQL.InsertStartLog(), param);

            // 自動採番されたid, configdata取得
            IEnumerable<BatchConfigDTO> configs =
                await _posgre.Select<BatchConfigDTO>(BatchLogSQL.GetConfig(), param);

            return configs.Single();
        }

        public async Task UpdateError(Exception ex, BatchConfigDTO config)
        {
            var param = new DynamicParameters();
            param.Add("status", "ERROR");
            param.Add("message", $"{ex.Message}\n{ex.StackTrace}");
            param.Add("log_id", config.LogId);

            await _posgre.Execute(BatchLogSQL.UpdateStatus(), param);
        }

        public async Task UpdateStatus(BatchConfigDTO config, TimeSpan timeSpan)
        {
            var param = new DynamicParameters();

            param.Add("log_id", config.LogId);
            param.Add("message", "");

            if (config.TimeoutMinutes < timeSpan.TotalMinutes)
            {   
                param.Add("status", "TIMEOUT");
            }
            else if (config.ExpectedDurationMinutes < timeSpan.TotalMinutes)
            {
                param.Add("status", "SLOW");
            }
            else
            {
                param.Add("status", "SUCCESS");
            }
            await _posgre.Execute(BatchLogSQL.UpdateStatus(), param);
        }
    }
}

File: batch/goBatch/internal/batchLogger/service/batchLogger.go
package service

import (
	"time"

	"github.com/kazGear/portfolio/goBatch/internal/batchLogger/model"
	"github.com/kazGear/portfolio/goBatch/internal/batchLogger/repository"
)

type BatchLogger struct {
    repository repository.BatchLoggerRepository
}

func NewBatchLogger(repository repository.BatchLoggerRepository) *BatchLogger {
    return &BatchLogger{ repository: repository }
}

func (b *BatchLogger) InsertStartLog(batchName string) (*model.BatchConfig, error) {
    config, err := b.repository.InsertStartLog(batchName)

    if err != nil {
        return nil, err
    }
    return config, nil
}

func (b *BatchLogger) UpdateError(config *model.BatchConfig, err error) error {
    err = b.repository.UpdateError(config, err)

    if err != nil {
        return err
    }
    return nil
}

func (b *BatchLogger) UpdateStatus(config *model.BatchConfig, timeSpan *time.Duration) error {
    err := b.repository.UpdateStatus(config, timeSpan)

    if err != nil {
        return err
    }
    return nil
}